Fixed-size notes support

qmaldon opened the pull request:
Hi all,

These changes apply to **stickynotes only.**

This is my first PR on a Mate repo. Please bear with me, since there are a few things that are not straightforward:

1) Multiple changes were done automatically by the build system into .po files.
    I understand these are translation files caused by a change I did in stickynotes/sticky-notes-preferences.ui
2) All changes were verified manually in my host (RHEL 8). I'm not sure if your CI/CD system will kick in during PR.
3) Is there a way to get this branch backported to 1.26?

**What is being enhanced & fixed:**
1) [Enhanced] Support for fixed size notes. That is, a mechanism to disallow notes resizing (on all notes).
2) [Bug fix] On a large note (vertical length > screen height), after restarting the stickynotes applet there are no verifications of the w,h sizes stored in the xml file; as consequence no automatic scrollbar is set, leading to an ugly and difficult to manage note.
  This is fixed now, since during startup a check based on screen size is done (propagated from existed code).

Please look at the following images to get an idea of the code changes:

![Screenshot from 2022-09-04 20-09-07](https://user-images.githubusercontent.com/1594532/188353435-de8f8fe3-6519-4c5a-819d-846e2a86e166.png)

![Screenshot from 2022-09-04 20-09-27](https://user-images.githubusercontent.com/1594532/188353466-b6df36aa-8275-4196-925d-fdbb6a5411f2.png)

I decided to make these changes since the stickynotes applet has been one of my main productivity tools since 2014 (I used to run RHEL 5 at that time).

Please let me know if there is any change needed or edit as you see fit.

Thanks
--
Quintin Maldonado

mbkma commented:
Hi, thanks for contributing! Please do not change po files here. This will be overwritten by transifex anyway. To untrack po files, I belive it is
`git rm -r --cached <folder>`
I am not sure though.

qmaldon committed:
stickynotes: add fixed-size notes support

Add a new preference to force fixed size on all notes to prevent
resizing. Also fix a bug where notes taller than the screen height would
not display scrollbars after applet restart.

Fixes #370

qmaldon commented:
> Hi, thanks for contributing! Please do not change po files here. This will be overwritten by transifex anyway. To untrack po files, I belive it is `git rm -r --cached <folder>` I am not sure though.

Hi @mbkma ,
Removed .po changes from branch as suggested. Please have a look.

I think there is an issue in the debian build. Fedora looks good.

> Is there a way to get this branch backported to 1.26?
Following up on this, perhaps it would be 1.26.2?

Thanks
--
Quintin Maldonado

zhuyaliang commented on stickynotes/stickynotes.c:
```diff
@@ -750,6 +752,46 @@ stickynote_set_visible (StickyNote *note,
+{
+    note->force_fixed_size = force_fixed_size;
+
+    if (force_fixed_size) {
```
There is a problem with the judgment here

zhuyaliang commented on stickynotes/stickynotes.c:
```diff
@@ -251,6 +251,8 @@ stickynote_new_aux (GdkScreen *screen,
     stickynote_set_font (note, NULL, TRUE);
     stickynote_set_locked (note, FALSE);
 
+    stickynote_set_fixed_size(note, stickynotes->force_fixed_size);
```
Add spaces and unify the code format

vkareh reviewed (approved):
I rebased from `master`, fixed the merge conflicts and the indentation/spacing issues. It works well for me and fixes the notes resizing automatically when you add text to them (which is really annoying).
